src/main/java/com/zendesk/maxwell/producer/MaxwellKinesisProducer.java

```diff
@@ -51,7 +53,11 @@ public void onFailure(Throwable t) {
 
 		logger.error("Exception during put", t);
 
-		cc.markCompleted();
+		if (!context.getConfig().ignoreProducerError) {
+			context.terminate(new RuntimeException(t));
```

Comment:
Hmm, this will change all the major methods in Maxwell to throw `Throwable` instead of `Exception`. Doesn't seem right.